applications/asset_tracker/src/env_sensors/env_sensors.h

```diff
@@ -82,13 +84,42 @@ int env_sensors_get_pressure(env_sensor_data_t *sensor_data);
  */
 int env_sensors_get_air_quality(env_sensor_data_t *sensor_data);
 
-
 /**
  * @brief Initialize and start the environmental sensors.
  *
  * @return 0 if the operation was successful, otherwise a (negative) error code.
  */
-int env_sensors_init_and_start(void);
+int env_sensors_init_and_start(const env_sensors_data_ready_cb cb);
+
+/**
+ * @brief Set environmental sensor's poll/send interval.
+ *
+ * @param interval_s Interval, in seconds. 0 to disable.
+ *
+ */
+void env_sensors_set_send_interval(const s32_t interval_s);
+
+/**
+ * @brief Get environmental sensor's poll/send interval.
+ *
+ * @return Interval, in seconds.
+ */
+s32_t env_sensors_get_send_interval(void);
+
+/**
+ * @brief Enable or disable back-off delay for sending environmental data.
+ *
+ * @param backoff_enable True to enable back-off delay, otherwise false.
+ *
+ */
+void env_sensors_set_backoff_enable(const bool backoff_enable);
```

Comment:
It's rarely seen to declare non-pointer arguments with `const` parameter, is there a good reason to do this?

Reply:
the function does not need to modify the input parameter.

Reply:
IMO it brings little to no value to the API readability. The function operates on a copy of the argument, so it doesn't really matter whether it modifies it or not.

Other story is with pointers - const pointer is a guarantee to the caller, that his data (pointed by the pointer) will not be modified (though in C nothing really is guaranteed, a nasty function could always drop the `const` by casting).